src/printf/printf.c

```diff
@@ -1282,22 +1282,26 @@ static inline void format_string_loop(output_gadget_t* output, const char* forma
       }
 #if PRINTF_SUPPORT_DECIMAL_SPECIFIERS
       case 'f' :
-      case 'F' :
+      case 'F' : {
+        floating_point_t va_float = (floating_point_t) (flags & FLAGS_LONG_LONG ? va_arg(args, long double) : va_arg(args, double));
```

Comment:
This is wrong... if the user passed a double, and we interpret it as a long double, the value will get messed up. Perhaps you tested on a platform where `double` and `long double` are the same type?

Reply:
%lf should passed long double into it, that's why %lf needed.
If %lf is used but user provide `double` or `float`, then it's a wrong usage and should
be messed up